homeassistant/components/watttime/sensor.py

```diff
@@ -0,0 +1,142 @@
+"""Support for WattTime sensors."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import TYPE_CHECKING
+
+from homeassistant.components.sensor import (
+    STATE_CLASS_MEASUREMENT,
+    SensorEntity,
+    SensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    ATTR_ATTRIBUTION,
+    ATTR_LATITUDE,
+    ATTR_LONGITUDE,
+    PERCENTAGE,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+)
+
+from .const import (
+    CONF_BALANCING_AUTHORITY,
+    CONF_BALANCING_AUTHORITY_ABBREV,
+    CONF_BALANCING_AUTHORITY_ID,
+    DATA_COORDINATOR,
+    DOMAIN,
+)
+
+ATTR_BALANCING_AUTHORITY = "balancing_authority"
+ATTR_BALANCING_AUTHORITY_ABBREV = "balancing_authority_abbreviation"
+ATTR_BALANCING_AUTHORITY_ID = "balancing_authority_id"
+
+DEFAULT_ATTRIBUTION = "Pickup data provided by WattTime"
+
+SENSOR_TYPE_REALTIME_EMISSIONS_MOER = "realtime_emissions_moer"
+SENSOR_TYPE_REALTIME_EMISSIONS_PERCENT = "realtime_emissions_percent"
+
+
+@dataclass
+class RealtimeEmissionsSensorDescriptionMixin:
+    """Define an entity description mixin for realtime emissions sensors."""
+
+    data_key: str
+
+
+@dataclass
+class RealtimeEmissionsSensorEntityDescription(
+    SensorEntityDescription, RealtimeEmissionsSensorDescriptionMixin
+):
+    """Describe a realtime emissions sensor."""
+
+
+REALTIME_EMISSIONS_SENSOR_DESCRIPTIONS = (
+    RealtimeEmissionsSensorEntityDescription(
+        key=SENSOR_TYPE_REALTIME_EMISSIONS_MOER,
+        name="Marginal Operating Emissions Rate",
+        icon="mdi:blur",
+        native_unit_of_measurement="lbs CO2/MWh",
+        state_class=STATE_CLASS_MEASUREMENT,
+        data_key="moer",
+    ),
+    RealtimeEmissionsSensorEntityDescription(
+        key=SENSOR_TYPE_REALTIME_EMISSIONS_PERCENT,
+        name="Relative Marginal Emissions Intensity",
+        icon="mdi:blur",
+        native_unit_of_measurement=PERCENTAGE,
+        state_class=STATE_CLASS_MEASUREMENT,
+        data_key="percent",
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up WattTime sensors based on a config entry."""
+    coordinator = hass.data[DOMAIN][DATA_COORDINATOR][entry.entry_id]
+    async_add_entities(
+        [
+            RealtimeEmissionsSensor(coordinator, description)
+            for description in REALTIME_EMISSIONS_SENSOR_DESCRIPTIONS
+            if description.data_key in coordinator.data
```

Comment:
Why check this? Will they always be in the result, even if temporarily unavailable ?

Reply:
Because the type of account you have determines whether one or both of those data points will be returned. For example, unless I have a Pro account, my data will never include the `moer` value.